pkg/estimator/client/accurate.go

```diff
@@ -86,6 +89,62 @@ func (se *SchedulerEstimator) GetUnschedulableReplicas(
 	})
 }
 
+func (se *SchedulerEstimator) maxAvailableComponentSets(
+	ctx context.Context,
+	cluster string,
+	namespace string,
+	components []workv1alpha2.Component,
+) (int32, error) {
+	client, err := se.cache.GetClient(cluster)
+	if err != nil {
+		return 0, err
+	}
+
+	pbReq := &pb.MaxAvailableComponentSetsRequest{
+		Cluster:    cluster,
+		Components: make([]pb.Component, 0, len(components)),
+	}
+
+	for _, comp := range components {
+		// Deep-copy so that pointer is not shared between goroutines
+		var cr *workv1alpha2.ComponentReplicaRequirements
+		if comp.ReplicaRequirements != nil {
+			cr = comp.ReplicaRequirements.DeepCopy()
+		}
+
+		pbReq.Components = append(pbReq.Components, pb.Component{
+			Name:                comp.Name,
+			Replicas:            comp.Replicas,
+			ReplicaRequirements: toPBReplicaRequirements(cr, namespace),
+		})
+	}
+
+	res, err := client.MaxAvailableComponentSets(ctx, pbReq)
+	if err != nil {
+		return 0, fmt.Errorf("gRPC request cluster(%s) estimator error when calling MaxAvailableComponentSets: %v", cluster, err)
+	}
+	return res.MaxSets, nil
+}
+
+// toPBReplicaRequirements converts the API ComponentReplicaRequirements to the pb.ReplicaRequirements value.
+func toPBReplicaRequirements(cr *workv1alpha2.ComponentReplicaRequirements, namespace string) pb.ReplicaRequirements {
+	var out pb.ReplicaRequirements
+	out.Namespace = namespace
+	if cr == nil {
+		return out
+	}
+	out.ResourceRequest = cr.ResourceRequest
+	out.PriorityClassName = cr.PriorityClassName
+	if cr.NodeClaim != nil {
+		out.NodeClaim = &pb.NodeClaim{
+			NodeAffinity: cr.NodeClaim.HardNodeAffinity,
+			NodeSelector: cr.NodeClaim.NodeSelector,
+			Tolerations:  cr.NodeClaim.Tolerations,
+		}
+	}
+	return out
+}
```

Comment:
This is correct and not a blocker. But we should give it another look at the definition of the gRPC data structure. The conversion should be more straightforward. @seanlaii

Reply:
PS:
I added a follow-up task to #6734 for revising the design here.